packages/frontend/index.d.ts

```diff
@@ -41,16 +41,18 @@ interface MoreType extends LinkType {
     more: true;
 }
 
-interface ReaderRevenueLink {
+interface ReaderRevenueCategories {
     contribute: string;
     subscribe: string;
     support: string;
 }
 
 interface ReaderRevenueLinks {
-    header: ReaderRevenueLink;
-    footer: ReaderRevenueLink;
-    sideMenu: ReaderRevenueLink;
+    header: ReaderRevenueCategories;
```

Comment:
I don't understand exactly the requirements here, but are the links necessarily tied to the position on the page? I'm a bit wary of tying the data model to a specific visual implementation unless it's absolutely needed if that makes sense.

Reply:
Yeah, I agree with the concern - I think realistically they are tied to a 'component' in that way though - from Frontend model https://github.com/guardian/frontend/pull/20958/files#diff-490210a7e352de3ddfc2d98c1db606dfR256 through to the reporting, every readerRevenue 'url' here has a reference to a position in the page. 

Do you have any thoughts on how we might abstract that? I don't know if creating an abstraction would make anything any more portable in terms of the data model as I guess that's the whole point of having the different links...

Reply:
Hey, thanks - the explanation helps. It sounds like introducing an abstraction will just confuse things here as you say.

I think a better medium-term solution is probably not to pass these in at all, but have some kind of config for dotcomponents that gets loaded at startup. I.e. rather than passing general config in with each request. But that's outside the scope of this PR!

Reply:
@nicl the thinking behind passing these across was that any updates made to the links in` `frontend` would then be available in `dotcom-rendering` without someone having to update two places. I like the idea of information like this (generally quite static data) being held in `config` rather than passed with every request. Perhaps a shared `config` between `frontend` and `dotcom-rendering` would address the problem of making (and knowing to make) updates in two places.